### DISCUSSION TITLE
TOTP no longer correct

### DISCUSSION THREAD

**Discussion by Tealk** [dani-garcia/vaultwarden#4292] [General]
Hello,

I have had the problem for a few days that the TOTP passwords are no longer correct.
They are not accepted by the websites and when I look in my Authenticator app the code is also different from the one in Vaultwarden.
If I then enter the URL for TOTP again, the code is generated correctly again.

**Comment by RealOrangeOne** [comment]
Is the clock on your device correct (down to the second)? TOTP is time-dependent, so if the time is adrift then the codes will be incorrect. TOTP codes are generated client-side, so it'll be the time of your device rather than of the server running Vaultwarden.

**Reply by Tealk** [reply]
As far as I can see, the time is correct on the Vautlwarden server as well as on the clients where the Bitwarden client is running.
The Bitwarden app on Android and the Authenticator app on the same device also show different values. Although the TOTP secret that is stored is the same, at least for the 2 entries I checked.

As soon as I enter the complete TOTP URL from the QR code, the code is correct. However, the secret has remained the same.

**Reply by BlackDex** [reply]
Seems like a client issue, not something server side fixable.
Vaultwarden only serve's the data, the clients handle all the rest.

**Reply by BlackRaph** [reply]
I have the exact same issue.
Even creating a new entry with a new TOTP doesn't work, the generated code isn't accepted by the website.
I'm running on Docker on a Synology NAS.
Time of the server is synch'd via ntp.

**Reply by stefan0xC** [reply]
@RRandaxhe have you checked the time on your client? E.g. by using sites like https://time.gov/ or https://uhr.ptb.de/ you can see how much the local device is deviating. The `/admin/diagnostics` also checks if the time is correct:
![grafik](https://github.com/user-attachments/assets/df04f1ac-ce09-4ddd-ae0f-86078967ad77)

**Reply by BlackRaph** [reply]
Thanks, issue was not on server but on my personal computer.  Don't know why, but windows stopped to synch my clock.  Issue solved.
